src/platform/forms/tests/validations.unit.spec.js

```diff
@@ -172,11 +172,11 @@ describe('Validations unit tests', () => {
       };
       const toDate = {
         day: {
-          value: '',
+          value: null,
```

Comment:
I think these values are initialized as empty strings, so the tests should probably keep them as empty strings.

Reply:
Maybe we can change the test to `expect(isValidDateRange(fromDate, toDate)).to.be.false;` instead. Since it's an invalid date, it won't validate a partial date, therefore, it should be false.

What do you think @cvalarida ?

Reply:
Baseed on:

```
export function dateToMoment(dateField) {
  return moment({
    year: dateField.year.value,
    month: dateField.month.value ? parseInt(dateField.month.value, 10) - 1 : '',
    day: dateField.day ? dateField.day.value : null,
  });
}
```

If there is an invalid date the default is `null` for day and `''` for month.

For this test, if I only change day to null, the test passes:

```
const toDate = {
        day: {
          value: null,
          dirty: true,
        },
        month: {
          value: '',
          dirty: true,
        },
        year: {
          value: '2008',
          dirty: true,
        },
```

Reply:
I'm still nervous about changing the test without understanding why it was passing before and why it's failing now. Also without understanding the way the forms library actually represents incomplete dates, I'm not sure changing the tests are the right answer. What if the library passes this "wrong" data to the validation function? Our test would pass, but it would fail in production. :confused: